src/custom_nodes/east_ocr/east_ocr.cpp

```diff
@@ -0,0 +1,399 @@
+
+    uint64_t byteSize = sizeof(float) * targetImageHeight * targetImageWidth * channels * outputBatch;
+
+    float* buffer = (float*)malloc(byteSize);
+
+    for (uint64_t i = 0; i < outputBatch; i++) {
+        cv::Size taretShape(targetImageWidth, targetImageHeight);
+        cv::Mat image = crop_and_resize(originalImage, boxes[i], taretShape);
+        if (convertToGrayScale) {
+            image = apply_grayscale(image);
+        }
+        if (imageLayout == "NCHW") {
+            auto imgBuffer = reorder_to_nchw((float*)image.data, image.rows, image.cols, image.channels());
+            std::memcpy(buffer + (i * channels * targetImageWidth * targetImageHeight), imgBuffer.data(), byteSize / outputBatch);
+        } else {
+            std::memcpy(buffer + (i * channels * targetImageWidth * targetImageHeight), image.data, byteSize / outputBatch);
+        }
+    }
+
+    output->data = reinterpret_cast<uint8_t*>(buffer);
+    output->dataBytes = byteSize;
+    output->dimsCount = 5;
+    output->dims = (uint64_t*)malloc(output->dimsCount * sizeof(uint64_t));
+    output->dims[0] = 1;
+    output->dims[1] = outputBatch;
+    if (imageLayout == "NCHW") {
+        output->dims[2] = channels;
+        output->dims[3] = targetImageHeight;
+        output->dims[4] = targetImageWidth;
+    } else {
+        output->dims[2] = targetImageHeight;
+        output->dims[3] = targetImageWidth;
+        output->dims[4] = channels;
+    }
+    output->precision = FP32;
+    return 0;
+}
+
+int extract_coordinates_into_output(struct CustomNodeTensor* output, const std::vector<cv::Rect>& boxes) {
+    uint64_t outputBatch = boxes.size();
+    uint64_t byteSize = sizeof(int32_t) * 4 * outputBatch;
+
+    int32_t* buffer = (int32_t*)malloc(byteSize);
+
+    for (uint64_t i = 0; i < outputBatch; i++) {
+        int32_t entry[] = {
+            boxes[i].x,
+            boxes[i].y,
+            boxes[i].width,
+            boxes[i].height};
+        std::memcpy(buffer + (i * 4), entry, byteSize / outputBatch);
+    }
+    output->data = reinterpret_cast<uint8_t*>(buffer);
+    output->dataBytes = byteSize;
+    output->dimsCount = 3;
+    output->dims = (uint64_t*)malloc(output->dimsCount * sizeof(uint64_t));
+    output->dims[0] = 1;
+    output->dims[1] = outputBatch;
+    output->dims[2] = 4;
+    output->precision = I32;
+    return 0;
+}
+
+int extract_scores_into_output(struct CustomNodeTensor* output, const std::vector<float>& scores) {
+    uint64_t outputBatch = scores.size();
+    uint64_t byteSize = sizeof(float) * outputBatch;
+
+    float* buffer = (float*)malloc(byteSize);
+    std::memcpy(buffer, scores.data(), byteSize);
+
+    output->data = reinterpret_cast<uint8_t*>(buffer);
+    output->dataBytes = byteSize;
+    output->dimsCount = 3;
+    output->dims = (uint64_t*)malloc(output->dimsCount * sizeof(uint64_t));
+    output->dims[0] = 1;
+    output->dims[1] = outputBatch;
+    output->dims[2] = 1;
```

Comment:
This will change with change in demultiplexer output formatting. Just be aware.

Reply:
Applied neccessary changes